### PR TITLE
ExportToCoco function mapping cat_ids to [index_base, index_base + nu…

### PR DESCRIPTION
**Pull request by ytzeng1**
add a mapping function and a parameter for users to set index base

### CODE REVIEW

**Comment by alexheat** [comment]
Thank you @ytzeng1 . Would you be able to provide a short explanation on what this will do?

**Comment by ytzeng1** [comment]
Hi @alexheat, this change addresses the cat_ids issues. First it will find unique cat_ids and sort them, and then create a dict where the keys are the origin cat_ids and the values are new assigned ids from [index_base, index_base + num_cats). And in the end use pandas map function to apply changes to the dataset before output. 

It's useful if the cat_ids are not continuous, especially for dataset subsets, or combined multiple datasets.

**Comment by alexheat** [comment]
That sounds great. I am going to check it out this weekend.

File: pylabel/exporter.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def ExportToYoloV5(self, output_path='training/labels', yaml_file='dataset.yaml'
 
         return output_file_paths
 
-    def ExportToCoco(self, output_path=None) -> List:
+    def ExportToCoco(self, output_path=None, index_base=1) -> List:
```
Comment (alexheat) [inline pylabel/exporter.py:363]: I will make an optional parameter.

**Review by alexheat — APPROVED** [review]
Looks good.

**Comment by alexheat** [comment]
Thank you! I will merge your change and then do some testing before I publish a new package.

**Comment by alexheat** [comment]
Hello @ytzeng1. I have release a new package (v0.1.21) with you code. Would you like to give it a try?  
You can see the release notes here
https://github.com/pylabel-project/pylabel/releases/tag/v0.1.21 

I put your code into a function and used it in several places, not only for the coco exporter. And I made some small changes to the naming, for example to call it 'cat_id_index' 

Thank you again for this very helpful addition. If you have any other ideas please let me know. 

Also, are you able to share an example of the code the that you use to train a Detector model? I would like to add that to the sample library.

**Comment by ytzeng1** [comment]
Hi @alexheat Thank you for implementing this feature so quickly. I confirm that the converted datasets work well on detectron2. I just switched from yolo to rcnn family of models, so my training code is pretty much straight from the tutorial. If you like, I can make another pull request on your sample library. Thanks again for working out a release in such a quick manner.

**Comment by alexheat** [comment]
Hi @ytzeng1 . Yes that would be awesome if you can contribute a sample of training with detectron2.

**Pull request merged**